consul/resource_consul_keys.go

```diff
@@ -154,7 +157,17 @@ func resourceConsulKeysCreateUpdate(d *schema.ResourceData, meta interface{}) er
 			}
 
 			flags := sub["flags"].(int)
-
+			cas := sub["cas"].(int)
+			if cas != -1 {
+				written, err := keyClient.Cas(path, value, flags, cas)
+				if err != nil {
+					return err
+				}
+				if written {
```

Comment:
what if written is false

Reply:
removed it. since if written is false still the key should not get removed